src/lib/stream.ts

```diff
@@ -0,0 +1,70 @@
+import { pipeline as pipelineSync } from 'node:stream'
+import { promisify } from 'node:util'
+
+import type { Handler, HandlerEvent, HandlerContext, StreamingHandler, StreamingResponse } from '../function/index.js'
+
+// Node v14 doesn't have node:stream/promises
+const pipeline = promisify(pipelineSync)
+
+declare global {
+  // eslint-disable-next-line @typescript-eslint/no-namespace
+  namespace awslambda {
+    function streamifyResponse(
+      handler: (event: HandlerEvent, responseStream: NodeJS.WritableStream, context: HandlerContext) => Promise<void>,
+    ): Handler
+
+    // eslint-disable-next-line @typescript-eslint/no-namespace
+    namespace HttpResponseStream {
+      function from(stream: NodeJS.WritableStream, metadata: Omit<StreamingResponse, 'body'>): NodeJS.WritableStream
+    }
+  }
+}
+
+/**
+ * Enables streaming responses. `body` accepts a `Readable` stream or Web stream.
```

Comment:
nit: I think this might be clearer to people.

```suggestion
 * Enables streaming responses. `body` accepts a Node.js `Readable` stream or a WHATWG `ReadableStream`.
```